### ISSUE TITLE
Landing Thrusters Slows and Eventually Softlocks the Ship

### ISSUE THREAD

**Issue by headwig75** [Malcolm-Q/LC-LateGameUpgrades#515]
On both a continuous and fresh save file, I've noticed that the ship, while maxed out in Landing Thrusters, takes off and lands slower and slower with each day. When I have to go to the Company Building, the ship will immediately spawn where it would normally land, then zips to where it normally spawns from and takes at least a full minute to land. The slowing effect worsens each day until the ship no longer lands before midnight; if the lever to take off is pulled in this state and the ship has not finished landing, it seems to prevent the night from automatically ending due to reaching midnight, which results in a soft lock. Exiting to the main menu and re-entering the game resets the speed of the ship but it is only a preventative measure, as eventually it reaches the same slowed speed.

**Comment by WhiteSpike** [comment]
I can't see how the logic for the upgrade manages to give you a slower speed.
```c#

        [HarmonyPatch(nameof(StartOfRound.ShipLeave))]
        [HarmonyPrefix]
        static void ShipLeavePrefix(StartOfRound __instance)
        {
            if (!UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_ENABLED) return;
            if (UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_AFFECT_LANDING && !UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_AFFECT_DEPARTING) 
            // if the speed was changed on landing and we want back to vanilla when leaving
            {
                // revert
                __instance.shipAnimator.speed /= LandingThrusters.GetLandingSpeedMultiplier();
                return;
            }
            // if the speed wasn't changed on landing and we want to change the speed when leaving
            if (!UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_AFFECT_LANDING && UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_AFFECT_DEPARTING)
                __instance.shipAnimator.speed *= LandingThrusters.GetLandingSpeedMultiplier(); // apply the speed buff
        }
        [HarmonyPatch(nameof(StartOfRound.EndOfGame))]
        [HarmonyPostfix]
        static void EndOfGamePostfix(StartOfRound __instance)
        {
            if (!UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_ENABLED) return;
            // if we didn't change the speed for leaving, there's nothing to do
            if (!UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_AFFECT_DEPARTING) return;
            // revert speed buff when reached orbit
            __instance.shipAnimator.speed /= LandingThrusters.GetLandingSpeedMultiplier();
        }

        [HarmonyPatch(nameof(StartOfRound.openingDoorsSequence))]
        [HarmonyPrefix]
        static void OpeningDoorsSequencePrefix(StartOfRound __instance)
        {
            if (!UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_ENABLED) return;
            // if we don't want to alter the speed when landing, do nothing
            if (!UpgradeBus.Instance.PluginConfiguration.LANDING_THRUSTERS_AFFECT_LANDING) return;
           // apply speed buff when landing
            __instance.shipAnimator.speed *= LandingThrusters.GetLandingSpeedMultiplier();
        }
```

``GetLandingSpeedMultiplier()`` always returns a value higher or equal to 1.

If your config doesn't have a special case in which I did not consider for the calculations, I genuinely don't know what's causing that behaviour.

**Comment by headwig75** [comment]
I'm honestly not sure either. As far as I can tell, I don't have any mods installed that should affect the landing speed of the ship, of all things. Maybe one of them randomly does, but I figured if there wasn't I should at least point it out just in case I did something lol. I got footage of this happening to better visualize it, but it's more or less as I described it; https://streamable.com/u88nct

If needed, I can provide my r2modman profile as a code. At least outwardly, there isn't one that should affect it, but if it helps I can provide it 🤷

**Comment by WhiteSpike** [comment]
Are you still having this issue?

If so, can you reproduce it and send me the logs related to that game session? 
You can also provide the used configuration for the issue to occur.

I have just returned from break thus not working on this mod for a while.

**Comment by headwig75** [comment]
I started a new save since then and didn't buy the upgrade out of uncertainty lol. I'll test it later to verify if it's still an issue and see what I find. My mods have changed a bit since I initially posted this but I'll report back when I test!

**Comment by headwig75** [comment]
So it seems to be the case that the issue ironed itself out. I tried it on the save I'd started using with no problems. I also tried installing GameMaster, which itself was causing other issues (and I also had installed at the time, so I assumed was a cause), but the upgrade still worked even then. I can only assume it was another mod I had installed but I can't recreate what was happening. Could be that the mod updated earlier today to 3.9.5, but I have no idea lol. I'll close the issue and open a new one if I find myself in the same situation later on
